sdk/rust/oak/src/lib.rs

```diff
@@ -580,7 +600,7 @@ pub fn run_command_loop<
 /// ```
 #[macro_export]
 macro_rules! entrypoint {
-    ($name:ident => $main_function:expr) => {
+    ($name:ident < $msg:ty > => $handler:expr) => {
```

Comment:
I believe that in the original PR for the entrypoint macro, we debated having a procedural macro for this instead. Now that we are adding more things here and the syntax gets more complicated, maybe we want to revisit that. With a fairly simple procedural macro, you could write it like this instead:
```
#[oak(node)]
fn my_entrypoint(receiver: Receiver<MyReceiver>) {
  ...
}

```

Another alternative that I think might work with just `macro_rules` is:
```
macro_rules! entrypoint {
  ($name:ident => |$receiver:ident : Receiver<$msg:ty>| $handler_body:expr) => { ... }
}
```

But maybe that last one is problematic because users may import `Receiver` differently.

Reply:
Fully agree with your points above, your current proposal seems to strike a good balance. 👍